cpp/tests/test_ide_secir_ageres.cpp

```diff
@@ -0,0 +1,159 @@
+    std::vector<mio::StateAgeFunctionWrapper> vec_delaydistrib2(num_transitions, delaydistribution2);
+    // TransitionDistribution is not used for SusceptibleToExposed. Therefore, the parameter can be set to any value.
+    vec_delaydistrib2[(int)mio::isecir::InfectionTransition::SusceptibleToExposed].set_distribution_parameter(-1.);
+    vec_delaydistrib2[(int)mio::isecir::InfectionTransition::InfectedNoSymptomsToInfectedSymptoms]
+        .set_distribution_parameter(2.0);
+
+    model.parameters.get<mio::isecir::TransitionDistributions>()[mio::AgeGroup(1)] = vec_delaydistrib2;
+
+    // Third AgeGroup for TransitionDistributions.
+    mio::SmootherCosine smoothcos3(2.5);
+    mio::StateAgeFunctionWrapper delaydistribution3(smoothcos3);
+    std::vector<mio::StateAgeFunctionWrapper> vec_delaydistrib3(num_transitions, delaydistribution3);
+    // TransitionDistribution is not used for SusceptibleToExposed. Therefore, the parameter can be set to any value.
+    vec_delaydistrib1[(int)mio::isecir::InfectionTransition::SusceptibleToExposed].set_distribution_parameter(-1.);
+    vec_delaydistrib1[(int)mio::isecir::InfectionTransition::InfectedNoSymptomsToInfectedSymptoms]
+        .set_distribution_parameter(4.0);
+
+    model.parameters.get<mio::isecir::TransitionDistributions>()[mio::AgeGroup(2)] = vec_delaydistrib3;
+
+    //TransitionProbabilities.
+    for (auto group = mio::AgeGroup(0); group < mio::AgeGroup(num_agegroups); ++group) {
+        std::vector<ScalarType> vec_prob(num_transitions, 0.5);
+        // The following probabilities must be 1, as there is no other way to go.
+        vec_prob[Eigen::Index(mio::isecir::InfectionTransition::SusceptibleToExposed)]        = 1;
+        vec_prob[Eigen::Index(mio::isecir::InfectionTransition::ExposedToInfectedNoSymptoms)] = 1;
+        model.parameters.get<mio::isecir::TransitionProbabilities>()[group]                   = vec_prob;
+    }
+
+    // Contact matrix.
+    mio::ContactMatrixGroup contact_matrix = mio::ContactMatrixGroup(1, num_agegroups);
+    contact_matrix[0] = mio::ContactMatrix(Eigen::MatrixXd::Constant(num_agegroups, num_agegroups, 10.));
+    model.parameters.get<mio::isecir::ContactPatterns>() = mio::UncertainContactMatrix(contact_matrix);
+
+    mio::ExponentialSurvivalFunction exponential(0.5);
+    mio::StateAgeFunctionWrapper prob(exponential);
+    for (auto group = mio::AgeGroup(0); group < mio::AgeGroup(num_agegroups); ++group) {
+        model.parameters.get<mio::isecir::TransmissionProbabilityOnContact>()[group] = prob;
+        model.parameters.get<mio::isecir::RelativeTransmissionNoSymptoms>()[group]   = prob;
+        model.parameters.get<mio::isecir::RiskOfInfectionFromSymptomatic>()[group]   = prob;
+    }
+    model.parameters.set<mio::isecir::Seasonality>(0.1);
+    // Start the simulation on the 40th day of a year (i.e. in February).
+    model.parameters.set<mio::isecir::StartDay>(40);
+
+    model.check_constraints(dt);
+
```

Comment:
```suggestion

mio::isecir::Simulation sim(model, dt);
sim.advance(tmax);
```
I find it cleaner to do one simulation here (insetad of two) and then do the comparisons.